akamai/cache-client_test.go

```diff
@@ -49,3 +54,92 @@ func TestConstructAuthHeader(t *testing.T) {
 	test.AssertNotError(t, err, "Failed to create authorization header")
 	test.AssertEquals(t, authHeader, expectedHeader)
 }
+
+type akamaiServer struct {
+	responseCode int
+}
+
+func (as *akamaiServer) akamaiHandler(w http.ResponseWriter, r *http.Request) {
+	var req struct {
+		Objects []string
+		Type    string
+		Action  string
+	}
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		fmt.Printf("Failed to read request body: %s\n", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	err = json.Unmarshal(body, &req)
+	if err != nil {
+		fmt.Printf("Failed to unmarshal request body: %s\n", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	fmt.Printf("Request: %#v\n", req)
+	var resp struct {
+		HTTPStatus       int
+		Detail           string
+		EstimatedSeconds int
+		PurgeID          string
+	}
+	resp.HTTPStatus = as.responseCode
+	resp.Detail = "?"
+	resp.EstimatedSeconds = 10
+	resp.PurgeID = "?"
+
+	for _, testURL := range req.Objects {
+		if !strings.HasPrefix(testURL, "http://") {
+			resp.HTTPStatus = http.StatusForbidden
```

Comment:
This is mean't to replicate the way Akamai responds in error cases. Instead of responding with a error code and a nil body they'll marshal a object that attempts to describe the issue as they see it.